diff --git a/dist/chunks/index.DzobfTyw.js b/dist/chunks/index.DzobfTyw.js
index 08c3c116fbf8fe4685b45dee57f0f645b1270d81..26f8e8f0f8e56cf4c5b08256746b66b727360f17 100644
--- a/dist/chunks/index.DzobfTyw.js
+++ b/dist/chunks/index.DzobfTyw.js
@@ -20498,7 +20498,8 @@ class Vitest {
 		this._fsCache = new FileSystemModuleCache(this);
 		this.snapshot = new SnapshotManager({ ...resolved.snapshotOptions });
 		this._resolver = new VitestResolver(this.viteConfig.cacheDir, resolved);
-		this._fetcher = createFetchModuleFunction(this._resolver, resolved, this._fsCache, this._traces, this._tmpDir);
+		const tmpDir = (this._tmpDir ??= join(tmpdir(), nanoid()));
+		this._fetcher = createFetchModuleFunction(this._resolver, resolved, this._fsCache, this._traces, tmpDir);
 	}
 	_restartPromise;
 	_restartQueued = false;
@@ -21433,10 +21434,19 @@ class Vitest {
 				for (const error of errors) this.logger.error("error during close", error);
 				this._checkUnhandledErrors(errors);
 			});
+			await this._clearTmpDir();
 			await this._traces?.finish();
 		})();
 		return this.closingPromise;
 	}
+	async _clearTmpDir() {
+		if (!this._tmpDir) return;
+		const tmpDir = this._tmpDir;
+		this._tmpDir = void 0;
+		try {
+			await rm(tmpDir, { recursive: true, force: true });
+		} catch {}
+	}
 	/**
 	* Closes all projects and exit the process
 	* @param force If true, the process will exit immediately after closing the projects.
